commit c943b3aa43e23c7aec0f6e7969b3c6eeee64311b
Author: Manoj Gupta <manojgupta@google.com>
Date:   Mon Apr 3 12:38:20 2023 -0700

    Revert "Fix code related to some recent changes in LLVM"
    
    This reverts commit 0aeedced03af8c6a8855c623199a8df4bbdbafdb.

diff --git a/addr2line.cc b/addr2line.cc
index 3979955..5d43261 100644
--- a/addr2line.cc
+++ b/addr2line.cc
@@ -98,13 +98,13 @@ void LLVMAddr2line::GetInlineStack(uint64_t address, SourceStack *stack) const {
     std::string dir_name;
     if (line_table->hasFileAtIndex(file)) {
       const auto &entry = line_table->Prologue.getFileNameEntry(file);
-      file_name = llvm::dwarf::toString(entry.Name).value();
+      file_name = llvm::dwarf::toString(entry.Name).getValue();
       if (entry.DirIdx > 0 &&
           entry.DirIdx <= line_table->Prologue.IncludeDirectories.size())
         dir_name =
             llvm::dwarf::toString(
                 line_table->Prologue.IncludeDirectories[entry.DirIdx - 1])
-                .value();
+                .getValue();
     }
     stack->emplace_back(function_name, dir_name, file_name, start_line, line,
                         discriminator);
diff --git a/llvm_propeller_profile_writer.cc b/llvm_propeller_profile_writer.cc
index 88dec6d..a98fbf3 100644
--- a/llvm_propeller_profile_writer.cc
+++ b/llvm_propeller_profile_writer.cc
@@ -318,8 +318,8 @@ bool PropellerProfWriter::Write(
     // picked by the compiler.
     for (auto &func_name : func_names) {
       symorder_stream << func_name.str();
-      if (cluster_id.has_value())
-        symorder_stream << ".__part." << *cluster_id;
+      if (cluster_id.hasValue())
+        symorder_stream << ".__part." << cluster_id.getValue();
       symorder_stream << "\n";
     }
   }
diff --git a/perfdata_reader.cc b/perfdata_reader.cc
index 97460c3..cca08f9 100644
--- a/perfdata_reader.cc
+++ b/perfdata_reader.cc
@@ -265,7 +265,7 @@ llvm::Optional<std::set<std::string>> FindFileNameInPerfDataWithFileBuildId(
     BinaryPerfInfo *info) {
   if (info->binary_info.build_id.empty()) {
     LOG(INFO) << "No Build Id found in '" << binary_file_name << "'.";
-    return std::nullopt;
+    return llvm::None;
   }
   LOG(INFO) << "Build Id found in '" << binary_file_name
             << "': " << info->binary_info.build_id;
@@ -277,7 +277,7 @@ llvm::Optional<std::set<std::string>> FindFileNameInPerfDataWithFileBuildId(
                 << "' has filename '" << fn << "'.";
     return buildid_names;
   }
-  return std::nullopt;
+  return llvm::None;
 }
 
 // Select mmaps from perf.data.
